api/mail/message.py

```diff
@@ -83,31 +85,135 @@ def __iadd__(self, attachment: mail.base.Attachment) -> _Message:
 
 
 _GroupSummariesT = T.Dict[idm.base.Group, persistence.GroupSummary]
-_WarnedSummariesT = T.Collection[T.Tuple[T.TimeDelta, _GroupSummariesT]]
 
 
-class NotificationEMail(_Message):
-    """ Notification e-mail """
-    _subject = "Action Required: HGI Vault Summary"
-    _template = resource.read_text("api.mail", "notification.j2")
+class DeletedEMail(_Message):
+    """ Notification e-mail for deleted files"""
+    _subject = "/lustre data deleted"
+    _template = resource.read_text("api.mail.templates", "deleted.j2")
+
+    def __init__(self, stakeholder: idm.base.User, group_summary: _GroupSummariesT,
+                 n_files: int, file_list: T.Dict,
+                 vault_documentation: str) -> None:
+        super().__init__(DeletedEMail.Context(stakeholder,
+                                              group_summary, n_files, file_list,
+                                              vault_documentation))
```

Comment:
personally, I'd define `Context` in `_Message` as an abstract class, i.e.
```python
@staticmethod
@abstractmethod
def Context(**kwargs: Dict[str, Any]) -> Dict[str, Any]:
    ...
```

and then define the `__init__` as
```python
def __init__(self, **kwargs: Dict[str, Any]) -> Dict[str, Any]:
    super().__init__(EmailClass.Context(**kwargs)
```

tidying it up even further would probably mean diving into metaclasses which is probably a bit over the top for this